src/Register.js

```diff
@@ -31,6 +33,14 @@ export class Register extends React.Component {
         }
         validateAll(data, rules, messages)
             .then(() => {
+                this.setState({errors: {}});
+                axios.post("http://httpbin.org/post", this.state)
```

Comment:
FYI: in the previous line, you call `setState`, but in this line, you have the old state, because actually in the current call of this function the link of the `this.state` was not changed.

Reply:
What is the best practice to send information about user to server if I don't want to list all fields?

Reply:
You can simply add new fields for this data in the state:
```js
state = {
user: {email: '', password: ''},
errors: {}
}
```
If you can not do that by any reasons you can just take fields you need:
```js
const {email, password} = this.state
post('...', {email, password})
```
You can write util metods for it:
```js
const pick = (obj, fields = []) => fields.reduce((result, field) => {result[field]=obj[field]; return result;}, {})
post('...', pick(this.state, ['email', 'password']);
```
in `lodash` there is already implemented method `pick`